### PR TITLE
New ggplot2 Cheatsheet

### PR DESCRIPTION
**Pull request by mr-karan**
## Description of new Instant Answer, or changes

Cheatsheet for http://ggplot2.org/

Screenshot: 

image

## Related Issues and Discussions

Fixes https://github.com/duckduckgo/zeroclickinfo-goodies/issues/3754

---

Instant Answer Page: https://duck.co/ia/view/ggplot2_cheat_sheet

### CODE REVIEW

**Comment by daxtheduck** [comment]
@mr-karan Thanks for the pull request! Unfortunately, we're only accepting contributions and improvements to a limited portion of the search space.
            
The staff and community are currently focused on making DuckDuckGo the [best search engine for programmers](https://forum.duckduckhack.com/t/duckduckhack-programming-mission-overview/53). To ensure we're making as much progress as possible, **we are only accepting Instant Answers and updates related to the programming mission**.
    
If this is not related to the Programming Mission, it will likely be closed but you can:
    
1. Help us prevent further confusion by telling us where you found out about DuckDuckHack or where we could've alerted you earlier.
2. Keep this code handy! We'll be expanding into other search topics in the future!
3. Join us at our new home: https://forum.duckduckhack.com/

If this is related to the Programming Mission, it will be reviewed by staff and the community as soon as possible!

**Comment by daxtheduck** [comment]
Metadata placeholder post
            
*If you haven't already, please edit your first comment to include a link to the IA page at the bottom.*
            
---
            
**How to find your IA page**
- New Instant Answer
  -- follow the steps [here](https://duck.co/ia/new_ia) to create your IA page.
              
- Existing Instant Answer
  -- find your IA page from this [list](https://duck.co/ia)

**Comment by daxtheduck** [comment]
Automated screenshot from [beta.duckduckgo.com](https://beta.duckduckgo.com/?q=ggplot2 cheatsheet)
                
                [Mobile screenshot](https://images.duckduckgo.com/iu/?u=https://ia-screenshots.s3.amazonaws.com/ggplot2_cheatsheet_mobile.png?WEmnfwe7Ri)

**Comment by daxtheduck** [comment]
Automated screenshot from [beta.duckduckgo.com](https://beta.duckduckgo.com/?q=ggplot2 cheatsheet)
                
                [Mobile screenshot](https://images.duckduckgo.com/iu/?u=https://ia-screenshots.s3.amazonaws.com/ggplot2_cheatsheet_mobile.png?fAXwwcz1lC)

**Comment by daxtheduck** [comment]
Automated screenshot from [beta.duckduckgo.com](https://beta.duckduckgo.com/?q=ggplot2 cheatsheet)
                
                [Mobile screenshot](https://images.duckduckgo.com/iu/?u=https://ia-screenshots.s3.amazonaws.com/ggplot2_cheatsheet_mobile.png?Rdugj5UGU0)

**Comment by mr-karan** [comment]
Any updates ? @tagawa

File: share/goodie/cheat_sheets/json/ggplot2.json [hunk 1]
```diff
@@ -0,0 +1,184 @@
+{
+	"id": "ggplot2_cheat_sheet",
+	"name": "Ggplot2",
```
Comment (tagawa) [inline share/goodie/cheat_sheets/json/ggplot2.json]: Please change this name to all lowercase: `ggplot2`

File: share/goodie/cheat_sheets/json/ggplot2.json [hunk 2]
```diff
@@ -0,0 +1,184 @@
...
+	"description": "ggplot2 for R Cheatsheet",
+
+	"metadata": {
+		"sourceName": "ggplot2 Official Docs",
+		"sourceUrl": "http://docs.ggplot2.org/current/"
+	},
+
+	"aliases": [
```
Comment (tagawa) [inline share/goodie/cheat_sheets/json/ggplot2.json]: Please remove this aliases array.

File: share/goodie/cheat_sheets/json/ggplot2.json [hunk 3]
```diff
@@ -0,0 +1,184 @@
+{
+	"id": "ggplot2_cheat_sheet",
+	"name": "Ggplot2",
+	"description": "ggplot2 for R Cheatsheet",
+
+	"metadata": {
+		"sourceName": "ggplot2 Official Docs",
```
Comment (tagawa) [inline share/goodie/cheat_sheets/json/ggplot2.json]: Please change this source name to all lowercase: `ggplot2 official docs`

**Comment by tagawa** [comment]
@mr-karan Looks like we haven't had any community reviews so I added a few comments above.

**Comment by mr-karan** [comment]
fixed 👍  @tagawa

**Comment by daxtheduck** [comment]
Automated screenshot from [beta.duckduckgo.com](https://beta.duckduckgo.com/?q=ggplot2 cheatsheet)
                
                [Mobile screenshot](https://images.duckduckgo.com/iu/?u=https://ia-screenshots.s3.amazonaws.com/ggplot2_cheatsheet_mobile.png?ataamMGEGm)

**Comment by daxtheduck** [comment]
Automated screenshot from [beta.duckduckgo.com](https://beta.duckduckgo.com/?q=ggplot2 cheatsheet)
                
                [Mobile screenshot](https://images.duckduckgo.com/iu/?u=https://ia-screenshots.s3.amazonaws.com/ggplot2_cheatsheet_mobile.png?bCrleGHnKC)

File: share/goodie/cheat_sheets/json/ggplot2.json [hunk 4]
```diff
@@ -0,0 +1,180 @@
+{
+	"id": "ggplot2_cheat_sheet",
+	"name": "ggplot2",
+	"description": "ggplot2 for R Cheatsheet",
```
Comment (tagawa) [inline share/goodie/cheat_sheets/json/ggplot2.json]: Sorry, one more - please could we change this to something more descriptive, such as "A guide to functions for ggplot2, a plotting system for R"

**Comment by daxtheduck** [comment]
Automated screenshot from [beta.duckduckgo.com](https://beta.duckduckgo.com/?q=ggplot2 cheatsheet)
                
                [Mobile screenshot](https://images.duckduckgo.com/iu/?u=https://ia-screenshots.s3.amazonaws.com/ggplot2_cheatsheet_mobile.png?9hH9o8jb8Y)

**Comment by mr-karan** [comment]
@tagawa for some reason I'm not able to see your older comment again, but I've made all the changes :)

**Comment by daxtheduck** [comment]
Automated screenshot from [beta.duckduckgo.com](https://beta.duckduckgo.com/?q=ggplot2 cheatsheet)
                
                [Mobile screenshot](https://images.duckduckgo.com/iu/?u=https://ia-screenshots.s3.amazonaws.com/ggplot2_cheatsheet_mobile.png?h4Bhm2d4Ra)

**Comment by daxtheduck** [comment]
Automated screenshot from [beta.duckduckgo.com](https://beta.duckduckgo.com/?q=ggplot2 cheatsheet)
                
                [Mobile screenshot](https://images.duckduckgo.com/iu/?u=https://ia-screenshots.s3.amazonaws.com/ggplot2_cheatsheet_mobile.png?xYGEMaPyv9)

**Comment by daxtheduck** [comment]
Automated screenshot from [beta.duckduckgo.com](https://beta.duckduckgo.com/?q=ggplot2 cheatsheet)
                
                [Mobile screenshot](https://images.duckduckgo.com/iu/?u=https://ia-screenshots.s3.amazonaws.com/ggplot2_cheatsheet_mobile.png?XE8UmUpwNi)

**Comment by daxtheduck** [comment]
Automated screenshot from [beta.duckduckgo.com](https://beta.duckduckgo.com/?q=ggplot2 cheatsheet)
                
                [Mobile screenshot](https://images.duckduckgo.com/iu/?u=https://ia-screenshots.s3.amazonaws.com/ggplot2_cheatsheet_mobile.png?WAr9PZZqVO)

**Comment by tagawa** [comment]
@moollaza Any idea what's happening here? I'm deploying to beta but the latest file changes, e.g. description, are not taking effect. The ID matches the IA page and the filename is fine.

**Comment by pjhampton** [comment]
Hmm, weird - there was a link wrong in the PR template. I'm not sure if this was the casue. This should be fixed now @tagawa @mr-karan

**Comment by pjhampton** [comment]
Many thanks @mr-karan 💙  💛  ❤️  💜

**Comment by pjhampton** [comment]
@tagawa are you happy for this to be merged?

**Comment by tagawa** [comment]
@pjhampton Yes, looks good to me now.
@mr-karan Thank you for making this and sorry it took so long to get sorted out. Let's merge and this should go live in a week or so.

**Comment by daxtheduck** [comment]
@mr-karan this is now merged! It should be live on DuckDuckGo within one week (or during the next release) and we'll be sure to notify you when it happens.

After your Instant Answer goes live for the World to use, you will be able to see [traffic information](http://docs.duckduckhack.com/maintaining/tools.html#monitor-your-instant-answer-page) about your IA as well as receive feedback directly from users. 
                
Please refer to the [maintenance guide and docs](http://docs.duckduckhack.com/maintaining/guidelines.html) to keep up with potential changes to your Instant Answer.
                
Thank you again for your help!

**Pull request merged**